lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -824,7 +824,8 @@ sub investigate {
     my $investigation = $job->investigate;
     $self->respond_to(
         json => {json => $investigation},
-        html => {text => join("\n", map { "$_: $investigation->{$_}" } keys %$investigation)});
+        html => {inline => join("\n", map { "<tr><td style=\"vertical-align: top;\"><b>$_:</b></td><td><pre>$investigation->{$_}</pre></td><tr>" } keys %$investigation)}
```

Comment:
Hopefully `$investigation->{$_}` doesn't contain `</pre>`.

Reply:
I don't understand. Why should it?

Reply:
Do you mean if `$investigation` would include anything that looks like HTML code that could destroy the document or have a malicious effect?

Reply:
This would probably make a XSS or CSRF attack possible. Haven't had a closer look though, what the values can possibly contain.

Reply:
yes, obviously, but

1. Didn't we have this possibility already in before?
2. One would need to bring in HTML tags over either job settings or git commit messages or filenames, possible but hard, still something we likely want to avoid
3. What to do?